sources/parsing/parse_command.c

```diff
@@ -0,0 +1,111 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   parse_command.c                                    :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: taretiuk <taretiuk@student.42.fr>          +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2024/10/15 16:11:49 by taretiuk          #+#    #+#             */
+/*   Updated: 2024/10/16 19:04:21 by taretiuk         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "parsing.h"
+
+static char	**split_excluding_quotes(char **result, char **s, int sz)
+{
+	int		i;
+	int		j;
+	int		res_i;
+	char 	**temp_split;
+	
+	i = 0;
+	res_i = 0;
+	while(s[i] != NULL)
+	{
+		temp_split = ft_split_ex(s[i++], ' ', &sz);
+		j = 0;
+		while (j < sz)
+		{
+			result[res_i++] = temp_split[j++];
+		}
+		free(temp_split);
+	}
+	result[res_i] = NULL;
+	return (result);
+}
+
+static void	calculate_spaces(char **s, int *sz, int *total_sz)
+{
+	int		i;
+    int		j;
+    char	**tok_space;
+
+	i = 0;
+	*total_sz = 0;
+	while (s[i] != NULL)
+	{
+		tok_space = ft_split_ex(s[i], ' ', sz);
+		if (tok_space == NULL)
+			return ;
+		*total_sz += *sz;
+		j = 0;
+		while (j < *sz)
+		{
+			free(tok_space[j++]);
+		}
+		free(tok_space);
+		i++;
+	}
+}
+
+char	**split_by_space(char **s)
+{
+	int		sz;
+	int		total_sz;
+	char	**result;
+	
+	calculate_spaces(s, &sz, &total_sz);
+	result = malloc(sizeof(char *) * (total_sz + 1));
```

Comment:
Please use ft_calloc_if every time you need some memory allocated in this project, we'll need this later for leak checks